pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TabularParticipantParser.java

```diff
@@ -152,11 +144,16 @@ private Map<String, Object> getDefForQuestion(ParticipantColumn column, Map<Stri
         if (activityDef == null) {
             return null;
         }
-        // find the question with the matching stableId
-        Map<String, Object> matchingDef = ((List<Map<String, Object>>) activityDef.get(ESObjectConstants.QUESTIONS))
-                .stream().filter(q -> q.get(ESObjectConstants.STABLE_ID).equals(column.getName()))
-                .findFirst().orElse(null);
-        return matchingDef;
+        // find the question with the matching stableId, and get the index
+        List<Map<String, Object>> questionList = ((List<Map<String, Object>>) activityDef.get(ESObjectConstants.QUESTIONS));
+        for (int i = 0; i < questionList.size(); i++) {
+            Map<String, Object> question = questionList.get(i);
+            if (question.get(ESObjectConstants.STABLE_ID).equals((column.getName()))) {
+                question.put("index", i);
+                return question;
+            }
+        }
+        return null;
```

Comment:
What do you think about having `Optional` pattern to avoid null returns and null checks? Method which has `List` type return value to return empty list and method which returns not `List` type value we return `Optional` type value, to think about this pattern and follow in general?

Reply:
To be honest, I'm not fluent in the optional pattern.  My sense here is that it won't save us on the logic checks, since we need totally different behavior if the question doesn't exist.  But again, not sure